controllers/machine_controller_noderef.go

```diff
@@ -85,6 +86,14 @@ func (r *MachineReconciler) reconcileNode(ctx context.Context, cluster *clusterv
 		r.recorder.Event(machine, corev1.EventTypeNormal, "SuccessfulSetNodeRef", machine.Status.NodeRef.Name)
 	}
 
+	// Reconcile node annotations.
+	annotations.SetNodeAnnotation(node, clusterv1.ClusterNameAnnotation, machine.Spec.ClusterName)
+	annotations.SetNodeAnnotation(node, clusterv1.ClusterNamespaceAnnotation, machine.GetNamespace())
+	annotations.SetNodeAnnotation(node, clusterv1.MachineAnnotation, machine.Name)
```

Comment:
Would it also be helpful to annotate the Node with the kind/name of the owner if a controller reference exists on the Machine here?

That would help link the Node more directly to a MachineSet or KubeadmControlPlane (if the Machine is managed by either of those).

Reply:
Added